webclient/src/views/view/view-view.js

```diff
@@ -168,32 +168,33 @@ navigatum.registerView('view', {
                 navigatum.getModule("interactive-map").then(function(c) {
                     _this.map.interactive.component = c;
 
-                    var map = _this.map.interactive.map;
+                    let map = _this.map.interactive.map;
+                    let marker = _this.map.interactive.marker;
                     // The map might or might not be initialized depending on the type
                     // of navigation.
                     if (document.getElementById("interactive-map")) {
-                        if (!document.getElementById("interactive-map")
-                                    .classList.contains("leaflet-container")) {
-                            document.getElementById("interactive-map").classList.remove("loading");
+                        if (document.getElementById("interactive-map").classList.contains("mapboxgl-map")) {
+                            marker.remove()
+                        }
+                        else {
                             map = c.initMap('interactive-map');
                             _this.map.interactive.map = map;
-                        } else {
-                            // Clean old data from the map
-                            map.eachLayer(function(layer) {
-                                if (layer instanceof L.Marker)
-                                    layer.remove();
-                            });
+
+                            document.getElementById("interactive-map").classList.remove("loading");
                         }
                     }
-
-                    // Use 17 as default zoom for now, TODO: Compute
-                    var coords = _this.view_data.coords;
-                    map.setView([coords.lat, coords.lon], 17, {
-                        // Only pan visibly when interactive map was displayed
-                        animate: from_map === "interactive",
-                    });
-
-                    L.marker([coords.lat, coords.lon], {icon: c.icon}).addTo(map)
+                    marker = c.initMarker();
+                    _this.map.interactive.marker = marker;
+                    const coords = _this.view_data.coords;
+                    marker.setLngLat([coords.lon, coords.lat]).addTo(map);
+                    // Use 16 as default zoom for now, TODO: Compute
+                    if (from_map === "interactive"){
+                        map.flyTo({center: [coords.lon, coords.lat], zoom: 16, duration: 5000});
```

Comment:
```suggestion
                        map.flyTo({
                            center: [coords.lon, coords.lat],
                            zoom: 16,
                            speed: 1,
                            maxDuration: 2000
                        });
```

To be honest I think we need to reduce the fly time. It looks nice on fast devices, but introduces quite some lag on slower mobile devices.